### ISSUE TITLE
🚀 Feature: Onboarding UI/UX

### ISSUE THREAD

**Issue by tenokami** [arc53/DocsGPT#1201]
### 🔖 Feature description

An onboarding experience that happens once for every new user.

Here is figma prototype: https://www.figma.com/design/OXLtrl1EAy885to6S69554/DocsGPT?node-id=4254-8745&t=92rsFCtdBQlNZGQ6-1

here is the figma file, you can find onboarding on the bottom: https://www.figma.com/design/OXLtrl1EAy885to6S69554/DocsGPT?node-id=0-1&t=5SZr1TQA6ekFGx5Q-1

Please make sure to add smooth transitions between steps. You can suggest similar alternatives for the loader animation to choose from and then proceed. Post results for review 

### 🎤 Why is this feature needed ?

To enhance UI/UX for the new users, leading to higher retention and conversion

### ✌️ How do you aim to achieve this?

By incorporating new user onboarding

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

**Comment by sanketshinde3001** [comment]
I'm interested

**Comment by sOnU1002** [comment]
i can do this task

**Comment by tenokami** [comment]
@sOnU1002 Hey. Thank you so much!

Today I plan to redesign the "upload file" modal which will affect this issue and more. Once I do, I will comment it. So please try to keep it in mind until then.

**Comment by tenokami** [comment]
@sOnU1002 Figma is updated and final (for now 😁)

**Comment by tenokami** [comment]
@sanketshinde3001 Don't hesitate to contact me for any questions and if you have something to show

**Comment by sanketshinde3001** [comment]
Sure 🧡

**Comment by RohittCodes** [comment]
@sanketshinde3001 are you still working on it?

**Comment by sanketshinde3001** [comment]
@RohittCodes yes.

**Comment by sanketshinde3001** [comment]
@tenokami  Yes, saw that already.👍

**Comment by RohittCodes** [comment]
@sanketshinde3001 still working mate?

**Comment by sanketshinde3001** [comment]
@RohittCodes Yes , ongoing brother.

**Comment by sanketshinde3001** [comment]
@tenokami should add multiple routes for those pages ? or Just keep them on same page ?

**Comment by tenokami** [comment]
@ManishMadan2882 Can you help with this question?

**Comment by ManishMadan2882** [comment]
Think it would be better to keep them on a single page.

**Comment by sanketshinde3001** [comment]
@tenokami https://docsgptui.vercel.app/ here is ongoing work demo. In 2-3 days i'll complete it with smooth animations and responsiveness. After that you can check it out and and give me modifications in it . And then i'll integrate it in main DOCSGPT code.

**Comment by tenokami** [comment]
@sanketshinde3001Thank you.

Feedback from me:
1. Move all content a little higher on the page, more natural for viewing (updated in figma too)
2. 
![image](https://github.com/user-attachments/assets/9905fb35-dc10-49f9-96f8-9c88d26cf068)
These should have a white background
3. there should be the title and logo from figma
![image](https://github.com/user-attachments/assets/1b5dbefa-125f-4d07-8666-3f87712365cc)

**Comment by tenokami** [comment]
![image](https://github.com/user-attachments/assets/20b2bb7c-d40e-4f12-9209-e7d7def9bd2f)
@sanketshinde3001 I'm not sure about this screen, is it the loading for uploading document or for training?

**Comment by sanketshinde3001** [comment]
@tenokami hii sir, those last two stages are not completed yet. I just added sample spinner just to make skeleton on ui. 
I'm working on those. 
I'll make all changes that you mentioned above soon.

**Comment by ManishMadan2882** [comment]
Hi @sanketshinde3001
Is there any update on this issue?

**Comment by sanketshinde3001** [comment]
@ManishMadan2882 Working on it sir. actually due to college workload , not get enough time for it. but after tomorrow mid day . I'll start work on it. and in 2 days will complete it and send for review.

**Comment by ManishMadan2882** [comment]
@sanketshinde3001 no worries at all, take your time.

**Comment by sanketshinde3001** [comment]
@ManishMadan2882 Thanks sir😄

**Comment by sanketshinde3001** [comment]
@ManishMadan2882
can you please review it ?

https://docsgptui.vercel.app/   
For dark mode click on that globe icon at top-left.

if found any issues then give me modifications in it 
That last spinner is not accurate as per figma file. I tried lot to mimic it but not able to do that :-\

**Comment by tenokami** [comment]
@sanketshinde3001 Hey!

Could you please work towards aligning the design more closely with the Figma reference? Reducing visual differences would be greatly appreciated, especially add smooth hover effects, selection states, and transitions between steps.

Additionally, it would be great if the color and shadow values could match, and if the title and subtitle could be styled on two lines as in Figma.

Please don’t hesitate to reach out if you need any additional information or clarification. Thank you!

**Comment by sanketshinde3001** [comment]
@tenokami sure sir.

**Comment by tenokami** [comment]
> @tenokami Sir , is it ok to make pr after hacktoberfest. I need some time due to diwali. I also want to make contribution docsgpt telegram bot repo.

@sanketshinde3001 Yes, of course! Very appreciated.

**Comment by sanketshinde3001** [comment]
@tenokami Hello sir , added smooth transition effects and hover effects. Plz review it once and  give me improvements in it.
till then i start working in the optimization of the code ..

**Comment by sanketshinde3001** [comment]
@tenokami Glad that u like it 😊. 
I'll update that color part and optimize the code .

**Comment by tenokami** [comment]
> @tenokami Glad that u like it 😊. 
> 
> I'll update that color part and optimize the code .
> 
> 

How about trying to place backgrounds like images in stead of trying to recreate them?

**Comment by sanketshinde3001** [comment]
> 7D54D1 (30% opacity)

![image](https://github.com/user-attachments/assets/468f0f6b-35b7-4a99-87ed-8a53e5969b63)

**Comment by sanketshinde3001** [comment]
@ManishMadan2882 Can you check this ?

**Comment by sanketshinde3001** [comment]
In codebase where should i add this ?

**Comment by tenokami** [comment]
> you mean that background color gradient ?

Sorry for late response, didn't get a notification. I meant these
<img width="1680" alt="Clean Screen UNAUTHORIZED" src="https://github.com/user-attachments/assets/0c362378-4fb5-4240-a612-44f8ab1c6908">
<img width="1680" alt="Onboarding Step 2" src="https://github.com/user-attachments/assets/23deee85-1df3-4f31-9db9-6690a8279f19">

**Comment by sanketshinde3001** [comment]
@tenokami 

![image](https://github.com/user-attachments/assets/fbe4d1fe-396f-4eeb-9d99-9c9062e57dcc)

how does it looks ?
changes are live on website...

**Comment by tenokami** [comment]
@sanketshinde3001 Looks good! Now would like some changes to the spinner.

Can we take this spinner which is used for document upload and change the red color to 7D54D1 ?

![image](https://github.com/user-attachments/assets/1a448b79-3c6a-4426-8d56-8f3e6b138ebc)

**Comment by sanketshinde3001** [comment]
Sure sir 👍

**Comment by sanketshinde3001** [comment]
@tenokami Done Sir .

**Comment by tenokami** [comment]
@sanketshinde3001 thanks!

could you also make these transparent with background blur like in figma? ![image](https://github.com/user-attachments/assets/77e25e0b-cc21-4a89-9a88-8cb2bd7b2ae3)

![image](https://github.com/user-attachments/assets/8982454c-d9fb-41ca-807b-61ebfd7992b7)

**Comment by sanketshinde3001** [comment]
Sure..
I was thinking same 😅.

**Comment by tenokami** [comment]
@sanketshinde3001 Yeah we can keep it like this.

**Comment by sanketshinde3001** [comment]
> An onboarding experience that happens once for every new user.

Sir, where should I include this UI component in the codebase?

**Comment by tenokami** [comment]
@sanketshinde3001 Sorry, this question is not for me

**Comment by sanketshinde3001** [comment]
Yes but 
Should i add whole flow as per above website
Or just replace already available components with new one?

**Comment by ManishMadan2882** [comment]
Yes, you can add the flow as per the above website, the flow is only for users that are not onboarded
No need to replace the available components, you can even reuse some of them.

**Comment by ManishMadan2882** [comment]
@sanketshinde3001 Are you working on this issue?

**Comment by sanketshinde3001** [comment]
![image](https://github.com/user-attachments/assets/4a73f59a-14ef-40eb-9b65-efad98b30d92)
@ManishMadan2882 Sir in frontend , it shows this Page directly , So i save 
'isonboarded' in local storage. 
only if it is true , it show Onboarding ui . 
working on -  Adding functionality to buttons (train) means connecting already present code with new ui.

is it enough ?

**Comment by ManishMadan2882** [comment]
Yes, you can do in this way.
Thanks for the update!

**Comment by sanketshinde3001** [comment]
Sir in docsgpt code , there are <img> tags to load image , should i convert Nextjs - <Image/>  tag into normal tag.

**Comment by ManishMadan2882** [comment]
Hey @sanketshinde3001
In the [frontend](https://github.com/arc53/DocsGPT/tree/main/frontend), we are using React and there is no Image component

**Comment by sanketshinde3001** [comment]
@ManishMadan2882 sir i added ui in website. I working fine. 
I tried to make those training functional. 
But facing some issues to do it.

**Comment by ManishMadan2882** [comment]
Push the changes and PR, so that we can review where is the issue
